src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -365,10 +375,11 @@ class BaseOptionsSelector extends Component {
                 label={this.props.textInputLabel}
                 accessibilityLabel={this.props.textInputLabel}
                 accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
-                onChangeText={this.props.onChangeText}
+                onChangeText={this.updateSearchValue}
+                errorText={this.state.errorMessage}
                 onSubmitEditing={this.selectFocusedOption}
                 placeholder={this.props.placeholderText}
-                maxLength={this.props.maxLength}
+                maxLength={this.props.maxLength + CONST.ERROR_EXCEED_RANGE}
```

Comment:
>E.g. maxLength = 500 + 20 will show the error from 501 to 520

Sorry I am not seeing what that means on your screenshot 😕 What do you mean by "will show the error from 501 to 520"? What and where will the "501 to 520" error show? 😕

Reply:
@flodnv 
 Sorry about wrong screenshot, this is the correct screenshot.
![Screenshot 2023-10-10 at 2 59 22 PM](https://github.com/Expensify/App/assets/41129870/670ddba2-0721-4a3a-857c-4ec628c84d9e)

```javascript
// the length can user type it without see the error with counter in the scrrenshot above.
this.props.maxLength = 500;

// the range we will display the error from 501 to 520
CONST.ERROR_EXCEED_RANGE = 20;

// so the max length that user can type at all and the input will allow it will be
maxLength={this.props.maxLength + CONST.ERROR_EXCEED_RANGE}
```

Reply:
So if I understand this correctly, it is impossible to type past 520 chars? What if I paste 1000 chars there?

Reply:
If we paste 1000 char, it will paste only 520 char, and this is the default behavior for maxLength, you can try [here](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_input_maxlength)